api4/channel_test.go

```diff
@@ -131,24 +114,17 @@ func TestCreateChannel(t *testing.T) {
 	CheckNoError(t, resp)
 
 	// Test posting Garbage
-	if r, err := Client.DoApiPost("/channels", "garbage"); err == nil {
-		t.Fatal("should have errored")
-	} else {
-		if r.StatusCode != http.StatusBadRequest {
-			t.Log("actual: " + strconv.Itoa(r.StatusCode))
-			t.Log("expected: " + strconv.Itoa(http.StatusBadRequest))
-			t.Fatal("wrong status code")
-		}
-	}
+	r, err := Client.DoApiPost("/channels", "garbage")
+	require.Error(t, err, "expected error")
```

Comment:
Sure thing!  
Unless I am reading it wrong, it would probably work either way.  Since type `model.AppError` is of type `error` (it implements the go builtin interface for `error` by exporting a `.Error() string` method), asserting the return is an Error should work, because `require.Error` simply checks that the value is both non-nil and is of type `error`.